clouddriver-api/src/main/java/com/netflix/spinnaker/clouddriver/orchestration/AnnotationsBasedAtomicOperationsRegistry.java

```diff
@@ -0,0 +1,168 @@
+              converters.size(), description, cloudProvider));
+    }
+    return (AtomicOperationConverter) converters.get(0);
+  }
+
+  @Override
+  public DescriptionValidator getAtomicOperationDescriptionValidator(
+      String validator, String cloudProvider) {
+    // Legacy naming convention which is not generic and validator name is specific to cloud
+    // provider
+    try {
+      DescriptionValidator descriptionValidator =
+          super.getAtomicOperationDescriptionValidator(validator, cloudProvider);
+      if (descriptionValidator != null) {
+        return descriptionValidator;
+      }
+    } catch (NoSuchBeanDefinitionException e) {
+    }
+
+    if (cloudProvider == null) return null;
+
+    Class<? extends Annotation> providerAnnotationType = getCloudProviderAnnotation(cloudProvider);
+
+    List validators =
+        applicationContext.getBeansWithAnnotation(providerAnnotationType).values().stream()
+            .filter(
+                (it) -> {
+                  Object validatorType =
+                      DefaultGroovyMethods.invokeMethod(
+                          it.getClass().getAnnotation(providerAnnotationType),
+                          "value",
+                          new Object[0]);
+                  return DescriptionValidator.getValidatorName((String) validatorType)
+                          .equals(validator)
+                      && it instanceof DescriptionValidator;
+                })
+            .collect(Collectors.toList());
+
+    return !validators.isEmpty() ? (DescriptionValidator) validators.get(0) : null;
+  }
+
+  protected Class<? extends Annotation> getCloudProviderAnnotation(String cloudProvider)
+      throws CloudProviderNotFoundException {
+    List<CloudProvider> cloudProviderInstances =
+        cloudProviders.stream()
+            .filter(it -> it.getId().equals(cloudProvider))
+            .collect(Collectors.toList());
+    if (cloudProviderInstances.isEmpty()) {
+      throw new CloudProviderNotFoundException(
+          String.format("No cloud provider named %s found", cloudProvider));
+    }
+    if (cloudProviderInstances.size() > 1) {
+      throw new RuntimeException(
+          String.format(
+              "More than one (%s) cloud providers found for the identifier '%s'",
+              cloudProviderInstances.size(), cloudProvider));
+    }
+    return cloudProviderInstances.get(0).getOperationAnnotationType();
+  }
+
+  private static class VersionedDescription {
+
+    private static final Splitter SPLITTER = Splitter.on("@");
+
+    @Nonnull String descriptionName;
+    @Nullable String version;
+
+    VersionedDescription(String descriptionName, String version) {
+      this.descriptionName = descriptionName;
+      this.version = version;
+    }
+
+    static VersionedDescription from(String descriptionName) {
+      if (descriptionName.contains("@")) {
+        List<String> parts = SPLITTER.splitToList(descriptionName);
```

Comment:
Think you should be able to do this without Guava with [String#split](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#split-java.lang.String-) though you'll get an array instead of a list.